### ISSUE TITLE
Docs/type for shouldForwardProp seems wrong

### ISSUE THREAD

**Issue by EloB** [styled-components/styled-components#3816]
From my research I see that `as` is missing on the docs and also @types/styled-components library.

`shouldForwardProp(prop, defaultValidatorFn, as)`.

I'm wrong?

**Comment by agriffis** [comment]
@EloB You're correct. The signature changed for v5 in https://github.com/styled-components/styled-components/pull/3451

The typings for v5 are maintained separately and I didn't submit a PR to update them. Please feel free to submit a PR over there. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/styled-components/index.d.ts

In v6 the signature changed again, see 6baaea485318946b9ef39b55eab9710fd22951a2. The typing is mostly correct in v6, but I've improved it slightly in https://github.com/styled-components/styled-components/pull/3809/commits/e95d07827d2cf09e5a6320c711d825505cacaf92 (not merged yet)

**Comment by quantizor** [comment]
Closing — this issue is stale (2+ years without resolution) and relates to an older version of styled-components. If this is still reproducible on the latest v6.x release, please open a new issue with a minimal reproduction.
